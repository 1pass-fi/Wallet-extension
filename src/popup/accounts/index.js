import React from 'react'
import './index.css'
import { Route, Switch } from 'react-router-dom'

import AccountHome from './accountHome'
import AccountImport from './accountImport'
import ImportPhrase from './importByPhrase/index'
import ImportFile from './importByFile/index'
import ImportFileSuccess from './importByFileSuccess/index'
import ImportPhraseSuccess from './importByPhraseSuccess/index'
import CreateWallet from './createWallet/index'
import Login from './accountLockScreen/index'

const Account = () => {
  return (
    <Switch>
      <Route exact path='/account'>
        <AccountHome />
      </Route>
      <Route path='/account/import/phrase/success'>
        <ImportPhraseSuccess />
      </Route>
      <Route path='/account/import/phrase'>
        <ImportPhrase />
      </Route>
      <Route path='/account/import/keyfile/success'>
        <ImportFileSuccess />
      </Route>
      <Route path='/account/import/keyfile'>
        <ImportFile />
      </Route>
      <Route path='/account/import'>
        <AccountImport />
      </Route>
      <Route path='/account/create'>
        <CreateWallet />
      </Route>
      <Route path='/account/login'>
        <Login />
      </Route>
    </Switch>
  )
}

export default Account
