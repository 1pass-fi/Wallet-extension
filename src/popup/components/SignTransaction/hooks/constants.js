export const TRANSACTION_TYPE = {
  ORIGIN_TOKEN_TRANSFER: 'ORIGIN_TOKEN_TRANSFER',
  CUSTOM_TOKEN_TRANSFER: 'CUSTOM_TOKEN_TRANSFER',
  CONTRACT_DEPLOYMENT: 'CONTRACT_DEPLOYMENT',
  CONTRACT_INTERACTION: 'CONTRACT_INTERACTION'
}

export const TRANSACTION_METHOD = {
  SET_APPROVAL_FOR_ALL: 'SET_APPROVAL_FOR_ALL',
  MINT_COLLECTIBLES: 'MINT_COLLECTIBLES',
  APPROVE: 'APPROVE',
  CONTRACT_INTERACTION: 'CONTRACT_INTERACTION',
  CONTRACT_DEPLOYMENT: 'CONTRACT_DEPLOYMENT',
  TOKEN_TRANSFER: 'TOKEN_TRANSFER'
}

export const TAB = {
  DETAIL: 'DETAIL',
  DATA: 'DATA',
  SIMULATION: 'SIMULATION'
}
