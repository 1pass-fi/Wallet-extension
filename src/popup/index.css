@tailwind base;
@tailwind components;
@tailwind utilities;

body {
  margin: 0;
  padding: 0;
  font-family: 'Sora', sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New', monospace;
}

html::-webkit-scrollbar {
  display: none;
}

/* width */
::-webkit-scrollbar {
  width: 4px;
  height: 4px;
}

/* Track */
::-webkit-scrollbar-track {
  background: rgba(165, 165, 165, 0);
  border-radius: 10px;
  border: 0px;
}

/* Handle */
::-webkit-scrollbar-thumb {
  background: #bef0ed;
  border-radius: 10px;
  border: 0px;
}

/* Handle on hover */
::-webkit-scrollbar-thumb:hover {
  background: #555;
}
