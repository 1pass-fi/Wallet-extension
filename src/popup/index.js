/* istanbul ignore next */
import React from 'react'
import ReactDOM from 'react-dom'
import * as browser from 'webextension-polyfill'
import { BrowserRouter as Router } from 'react-router-dom'
import { Provider } from 'react-redux'

import './index.css'
import Popup from './Popup'
import store from './store'

browser.runtime.sendMessage({ data: 'hello' })

ReactDOM.render(
  <Provider store={store}>
    <Router>
      <Popup />
    </Router>
  </Provider>,
  document.getElementById('root')
)
