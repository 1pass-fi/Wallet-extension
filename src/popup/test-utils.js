import React from 'react'
import { Provider } from 'react-redux'
import { Router } from 'react-router-dom'
import { render as rtlRender } from '@testing-library/react'
import { createMemoryHistory } from 'history'
import { applyMiddleware,createStore } from 'redux'
import thunk from 'redux-thunk'

import rootReducer from './reducers/index'

function render(
  ui,
  {
    history = createMemoryHistory(),
    initialState,
    store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    )
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

export * from '@testing-library/react'
export { render }
