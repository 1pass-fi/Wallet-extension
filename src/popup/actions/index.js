import { setActivities } from './activities'
import { setError } from './error'
import { importWallet } from './koi'

export default {
  setActivities,
  setError,
  importWallet
}
