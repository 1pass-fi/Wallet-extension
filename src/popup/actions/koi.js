import { setIsLoading } from './loading'
import { setError } from './error'
import { setCreateWallet } from './createWallet'
import { setAssets } from './assets'
import backgroundConnect, { CreateEventHandler } from './backgroundConnect'

import { MESSAGES, PATH } from 'constants'

import { SET_KOI } from 'actions/types'

export const importWallet = (inputData) => (dispatch) => {
  try {
    dispatch(setIsLoading(true))
    const { history, redirectPath } = inputData
    const importSuccessHandler = new CreateEventHandler(MESSAGES.IMPORT_WALLET_SUCCESS, response => {
      const { koiData } = response.data
      dispatch(setKoi(koiData))
      dispatch(setIsLoading(false))
      history.push(redirectPath)
    })
    const importFailedHandler = new CreateEventHandler(MESSAGES.ERROR, response => {
      console.log('=== BACKGROUND ERROR ===')
      const errorMessage = response.data
      dispatch(setIsLoading(false))
      dispatch(setError(errorMessage))
    })

    backgroundConnect.addHandler(importSuccessHandler)
    backgroundConnect.addHandler(importFailedHandler)
    backgroundConnect.postMessage({
      type: MESSAGES.IMPORT_WALLET,
      data: inputData
    })
  } catch (err) {
    dispatch(setError(err.message))
    dispatch(setIsLoading(false))
  }
}

export const loadWallet = (inputData) => (dispatch) => {
  try {
    dispatch(setIsLoading(true))
    const loadSuccessHandler = new CreateEventHandler(MESSAGES.LOAD_WALLET_SUCCESS, response => {
      const { koiData } = response.data
      dispatch(setKoi(koiData))
      dispatch(setIsLoading(false))
    })
    const loadFailedHandler = new CreateEventHandler(MESSAGES.ERROR, response => {
      console.log('=== BACKGROUND ERROR ===')
      const errorMessage = response.data
      dispatch(setIsLoading(false))
      dispatch(setError(errorMessage))
    })

    backgroundConnect.addHandler(loadSuccessHandler)
    backgroundConnect.addHandler(loadFailedHandler)
    backgroundConnect.postMessage({
      type: MESSAGES.LOAD_WALLET,
      data: inputData
    })
  } catch (err) {
    dispatch(setError(err.message))
    dispatch(setIsLoading(false))
  }
}

export const removeWallet = () => (dispatch) => {
  try {
    dispatch(setIsLoading(true))
    const removeSuccessHandler = new CreateEventHandler(MESSAGES.REMOVE_WALLET_SUCCESS, response => {
      const { koiData } = response.data
      dispatch(setKoi(koiData))
      dispatch(setIsLoading(false))
    })
    const removeFailedHandler = new CreateEventHandler(MESSAGES.ERROR, response => {
      console.log('=== BACKGROUND ERROR ===')
      const errorMessage = response.data
      dispatch(setIsLoading(false))
      dispatch(setError(errorMessage))
    })

    backgroundConnect.addHandler(removeSuccessHandler)
    backgroundConnect.addHandler(removeFailedHandler)
    backgroundConnect.postMessage({
      type: MESSAGES.REMOVE_WALLET
    })
  } catch (err) {
    dispatch(setError(err.message))
    dispatch(setIsLoading(false))
  }
}

export const lockWallet = (inputData) => (dispatch) => {
  try {
    const { history } = inputData
    dispatch(setIsLoading(true))
    const lockSuccessHandler = new CreateEventHandler(MESSAGES.LOCK_WALLET_SUCCESS, response => {
      dispatch(setIsLoading(false))
      history.push(PATH.LOGIN)
    })
    const lockFailedHandler = new CreateEventHandler(MESSAGES.ERROR, response => {
      console.log('=== BACKGROUND ERROR ===')
      const errorMessage = response.data
      dispatch(setIsLoading(false))
      dispatch(setError(errorMessage))
    })

    backgroundConnect.addHandler(lockSuccessHandler)
    backgroundConnect.addHandler(lockFailedHandler)
    backgroundConnect.postMessage({
      type: MESSAGES.LOCK_WALLET
    })
  } catch (err) {
    dispatch(setError(err.message))
    dispatch(setIsLoading(false))
  }
}

export const unlockWallet = (inputData) => (dispatch) => {
  try {
    const { history } = inputData
    dispatch(setIsLoading(true))
    const unlockSuccessHandler = new CreateEventHandler(MESSAGES.UNLOCK_WALLET_SUCCESS, response => {
      const { koiData } = response.data
      dispatch(setKoi(koiData))
      dispatch(setIsLoading(false))
      history.push('/account')
    })
    const unlockFailedHandler = new CreateEventHandler(MESSAGES.ERROR, response => {
      console.log('=== BACKGROUND ERROR ===')
      const errorMessage = response.data
      dispatch(setIsLoading(false))
      dispatch(setError(errorMessage))
    })

    backgroundConnect.addHandler(unlockSuccessHandler)
    backgroundConnect.addHandler(unlockFailedHandler)
    backgroundConnect.postMessage({
      type: MESSAGES.UNLOCK_WALLET,
      data: inputData
    })
  } catch (err) {
    dispatch(setError(err.message))
    dispatch(setIsLoading(false))
  }
}

export const generateWallet = (inputData) => (dispatch) => {
  try {
    const { stage, password } = inputData

    dispatch(setIsLoading(true))
    const generateSuccessHandler = new CreateEventHandler(MESSAGES.GENERATE_WALLET_SUCCESS, response => {
      const { seedPhrase } = response.data
      dispatch(setCreateWallet({ seedPhrase, stage, password }))
      dispatch(setIsLoading(false))
    })
    const generateFailedHandler = new CreateEventHandler(MESSAGES.ERROR, response => {
      console.log('=== BACKGROUND ERROR ===')
      const errorMessage = response.data
      dispatch(setIsLoading(false))
      dispatch(setError(errorMessage))
    })

    backgroundConnect.addHandler(generateSuccessHandler)
    backgroundConnect.addHandler(generateFailedHandler)
    backgroundConnect.postMessage({
      type: MESSAGES.GENERATE_WALLET,
      data: inputData
    })
  } catch (err) {
    dispatch(setError(err.message))
    dispatch(setIsLoading(false))
  }
}

export const saveWallet = (inputData) => (dispatch) => {
  try {
    const { history } = inputData
    dispatch(setIsLoading(true))
    const saveSuccessHandler = new CreateEventHandler(MESSAGES.SAVE_WALLET_SUCCESS, response => {
      const { koiData } = response.data
      dispatch(setKoi(koiData))
      dispatch(setIsLoading(false))
      history.push(PATH.HOME)
    })
    const saveFailedHandler = new CreateEventHandler(MESSAGES.ERROR, response => {
      console.log('=== BACKGROUND ERROR ===')
      const errorMessage = response.data
      dispatch(setIsLoading(false))
      dispatch(setError(errorMessage))
    })

    backgroundConnect.addHandler(saveSuccessHandler)
    backgroundConnect.addHandler(saveFailedHandler)
    backgroundConnect.postMessage({
      type: MESSAGES.SAVE_WALLET,
      data: inputData
    })
  } catch (err) {
    dispatch(setError(err.message))
    dispatch(setIsLoading(false))
  }
}

export const loadContent = () => (dispatch) => {
  try {
    dispatch(setIsLoading(true))
    const saveSuccessHandler = new CreateEventHandler(MESSAGES.LOAD_CONTENT_SUCCESS, response => {
      const { contentList } = response.data
      dispatch(setAssets(contentList))
      dispatch(setIsLoading(false))
    })
    const saveFailedHandler = new CreateEventHandler(MESSAGES.ERROR, response => {
      console.log('=== BACKGROUND ERROR ===')
      const errorMessage = response.data
      dispatch(setIsLoading(false))
      dispatch(setError(errorMessage))
    })

    backgroundConnect.addHandler(saveSuccessHandler)
    backgroundConnect.addHandler(saveFailedHandler)
    backgroundConnect.postMessage({
      type: MESSAGES.LOAD_CONTENT,
    })
  } catch (err) {
    dispatch(setError(err.message))
    dispatch(setIsLoading(false))
  }
}

export const setKoi = (payload) => ({ type: SET_KOI, payload })
