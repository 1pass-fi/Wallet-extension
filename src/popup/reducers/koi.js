import { SET_KOI } from 'actions/types'

const initialState = {
  arBalance: null,
  koiBalance: null,
  address: null
}

export default (state = initialState, action) => {
  const { type, payload } = action
  switch(type) {
    case SET_KOI:
      return { ...state, ...payload }
    default:
      return state
  }
}
