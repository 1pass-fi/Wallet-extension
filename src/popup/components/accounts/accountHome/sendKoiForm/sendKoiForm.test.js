import React from 'react'
import { makeTransfer } from 'actions/koi'
import { TYPE } from 'constants/accountConstants'
import { ERROR_MESSAGE, PATH } from 'constants/koiConstants'
import { createMemoryHistory } from 'history'
import rootReducer from 'popup/reducers'
import { act, fireEvent, render, screen } from 'popup/test-utils'
import { applyMiddleware,createStore } from 'redux'
import thunk from 'redux-thunk'

import '@testing-library/jest-dom/extend-expect'

import SendKoiForm from '.'

jest.mock('services/account', () => {
  return {
    popupAccount: {
      getAccount: jest.fn().mockReturnValue({
        get: {
          balance: jest.fn().mockReturnValue(12.22),
          koiBalance: jest.fn().mockReturnValue(12.22),
        },
      }),
    },
  }
})

jest.mock('actions/koi', () => ({
  makeTransfer: jest.fn(() => ({
    type: 'Fake action',
  })),
}))

describe('SendKoiForm component', () => {
  const accounts = [
    {
      type: TYPE.ARWEAVE,
      address: 'fakeAddress',
      accountName: 'Test account 1',
    },
    {
      type: TYPE.ETHEREUM,
      address: 'fakeAddress',
      accountName: 'Test account',
    },
  ]

  const validFormData = {
    accountName: 'Test account 1',
    selectedToken: 'AR',
    recipient: 'just-a-random-test-recipient',
    amount: 0.0023,
  }

  const chooseAccount = async (accountName) => {
    const accountDropdownBtn = screen.getAllByRole('button')[0]
    await act(async () => {
      await fireEvent.click(accountDropdownBtn)
      await fireEvent.click(screen.getByText(accountName))
    })
  }

  const chooseCurrency = async (selectedToken) => {
    const currencyDropdownBtn = screen.getAllByRole('button')[0]
    await act(async () => {
      await fireEvent.click(currencyDropdownBtn)
      await fireEvent.click(screen.getByText(selectedToken))
    })
  }

  const chooseRecipient = async (recipient) => {
    const recipientInput = screen.getByPlaceholderText(
      'Recipientâ€™s wallet address'
    )
    await act(async () => {
      await fireEvent.change(recipientInput, { target: { value: recipient } })
    })
  }

  const chooseAmount = async (amount) => {
    const amountInput = screen.getByRole('spinbutton')
    await act(async () => {
      await fireEvent.change(amountInput, { target: { value: amount } })
    })
  }

  const fillForm = async ({
    accountName,
    selectedToken,
    recipient,
    amount,
  }) => {
    await chooseAccount(accountName)
    await chooseCurrency(selectedToken)
    await chooseRecipient(recipient)
    await chooseAmount(amount)
  }

  const submitForm = async () => {
    const submitBtn = screen.getByRole('button', { name: /send/i })

    await act(async () => {
      fireEvent.click(submitBtn)
    })
  }

  let store
  let history
  beforeEach(() => {
    history = createMemoryHistory()

    store = createStore(
      rootReducer,
      {
        accounts,
      },
      applyMiddleware(thunk)
    )

    render(<SendKoiForm />, { store, history })
  })

  it('renders without crashing', () => {
    expect(screen.getAllByRole('textbox')).toHaveLength(2)
    expect(screen.getByRole('spinbutton')).toBeInTheDocument()
    expect(
      screen.getByRole('button', { name: /send token/i })
    ).toBeInTheDocument()
    expect(
      screen.getByText(/Make sure you have the correct wallet address/i)
    ).toBeInTheDocument()
  })

  describe('User fills form', () => {
    describe('User filled the account field', () => {
      it('shows token type field', async () => {
        expect(() => screen.getByPlaceholderText('Select token')).toThrow()

        const dropDownBtn = screen.getAllByRole('button')[0]
        await act(async () => {
          await fireEvent.click(dropDownBtn)
          await fireEvent.click(screen.getByText(accounts[0].accountName))
        })

        expect(screen.getByPlaceholderText('Select token')).toBeInTheDocument()
      })

      describe('User account is AR account', () => {
        it('shows token options of KOII and AR', async () => {
          expect(() => screen.getByPlaceholderText('Select token')).toThrow()

          const dropDownBtn = screen.getAllByRole('button')[0]
          await act(async () => {
            await fireEvent.click(dropDownBtn)
            await fireEvent.click(screen.getByText(accounts[0].accountName))
          })

          const currencyDropDownBtn = screen.getAllByRole('button')[0]
          await act(async () => {
            await fireEvent.click(currencyDropDownBtn)
          })

          expect(screen.getByText('AR')).toBeInTheDocument()
          expect(screen.getByText('KOII')).toBeInTheDocument()
          expect(() => screen.getByText('ETH')).toThrow()
        })
      })

      describe('User account is Ethereum account', () => {
        it('shows token options of ETH only', async () => {
          const dropDownBtn = screen.getAllByRole('button')[0]
          await act(async () => {
            await fireEvent.click(dropDownBtn)
            await fireEvent.click(screen.getByText(accounts[1].accountName))
          })

          const currencyDropDownBtn = screen.getAllByRole('button')[0]
          await act(async () => {
            await fireEvent.click(currencyDropDownBtn)
          })

          expect(screen.getByText('ETH')).toBeInTheDocument()

          expect(() => screen.getByText('AR')).toThrow()
          expect(() => screen.getByText('KOII')).toThrow()
        })
      })
    })

    describe('User chooses to edit account information', () => {
      it('shows back the choose account field and hide the choose token field', async () => {
        const dropDownBtn = screen.getAllByRole('button')[0]
        await act(async () => {
          await fireEvent.click(dropDownBtn)
          await fireEvent.click(screen.getByText(accounts[1].accountName))
        })

        const editBtn = screen.getByTestId('editBtn')
        await act(async () => {
          await fireEvent.click(editBtn)
        })

        expect(
          screen.getByPlaceholderText('Select your account')
        ).toBeInTheDocument()
        expect(() => screen.getByPlaceholderText('Select token')).toThrow()
      })
    })
  })

  describe('User fills then submits form', () => {
    describe('Valid form data', () => {
      it('shows transaction confirm modal', async () => {
        await fillForm({ ...validFormData, selectedToken: 'KOII' })

        await submitForm()
        expect(
          screen.getByText(/Transaction Confirmation/i)
        ).toBeInTheDocument()

        await act(async () => {
          await fireEvent.click(
            screen.getByRole('button', { name: 'No, Go Back' })
          )
        })

        setTimeout(() => {
          expect(() => screen.getByText(/Transaction Confirmation/i)).toThrow()
        }, 0)
      })

      describe('User chooses to make transfer', () => {
        describe('Transfer successfully', () => {
          it('calls makeTransfer and redirect to activity page', async () => {
            await fillForm(validFormData)

            await submitForm()
            expect(
              screen.getByText(/Transaction Confirmation/i)
            ).toBeInTheDocument()

            const sendBtnModal = screen.getAllByRole('button', {
              name: /send AR/i,
            })[1]

            await act(async () => {
              await fireEvent.click(sendBtnModal)
            })

            const userAccount = accounts[0]
            expect(makeTransfer).toHaveBeenCalledWith(
              {
                address: userAccount.address,
                type: userAccount.type,
                id: 0,
                label: userAccount.accountName,
                value: userAccount.accountName,
              },
              Number(validFormData.amount),
              validFormData.recipient,
              validFormData.selectedToken
            )

            // Need set time out to be truthy
            setTimeout(() =>
              expect(history.location.pathname).toBe(PATH.ACTIVITY)
            )
          })
        })
      })
    })

    describe('Invalid form data', () => {
      describe('Empty fields', () => {
        it('shows empty fields error message', async () => {
          await fillForm({ ...validFormData, recipient: '' })

          await submitForm()

          expect(() => screen.getByText(/Transaction Confirmation/i)).toThrow()
          expect(store.getState().error).toEqual(ERROR_MESSAGE.EMPTY_FIELDS)
        })
      })

      describe('Amount less than 0', () => {
        it('shows empty amount invalid message', async () => {
          await fillForm({ ...validFormData, amount: -5 })

          await submitForm()

          expect(() => screen.getByText(/Transaction Confirmation/i)).toThrow()
          expect(store.getState().error).toEqual(ERROR_MESSAGE.INVALID_AMOUNT)
        })
      })

      describe('Not selected account', () => {
        it('shows account not selected error message', async () => {
          await chooseRecipient(validFormData.recipient)
          await chooseAmount(validFormData.amount)

          await submitForm()

          expect(() => screen.getByText(/Transaction Confirmation/i)).toThrow()
          expect(store.getState().error).toEqual(ERROR_MESSAGE.SELECT_ACCOUNT)
        })
      })

      describe('Not selected token', () => {
        it('shows token not selected error message', async () => {
          await chooseAccount(validFormData.accountName)
          await chooseRecipient(validFormData.recipient)
          await chooseAmount(validFormData.amount)

          await submitForm()

          expect(() => screen.getByText(/Transaction Confirmation/i)).toThrow()
          expect(store.getState().error).toEqual(ERROR_MESSAGE.SELECT_TOKEN)
        })
      })

      describe('Amount is zero', () => {
        it('shows warning send zero Koi', async () => {
          await fillForm({ ...validFormData, amount: 0 })

          await submitForm()

          expect(store.getState().error).toEqual(ERROR_MESSAGE.SEND_ZERO_KOI)
        })
      })
    })
  })
})
