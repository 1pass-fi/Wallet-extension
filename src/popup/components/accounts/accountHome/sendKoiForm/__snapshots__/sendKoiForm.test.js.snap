// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for SendKoiForm component Render without crashing renders correctly 1`] = `
<div>
  <form
    class="send-koi-form"
  >
    <div
      class="koi-balance"
    >
      <span>
        Available balance: 
      </span>
      <b>
         KOI
      </b>
      <div
        class="amount-in-usd"
      >
        ~ $ 
        NaN
         USD
      </div>
    </div>
    <div
      class="recipient"
    >
      <div
        class="input-field form-input"
      >
        <div
          class="label"
        >
          To
        </div>
        <div
          class="input-wrapper"
        >
          <input
            class="input"
            name=""
            placeholder="Recipientâ€™s wallet address"
            type="text"
            value=""
          />
          <svg
            class="toggle-display"
          />
        </div>
      </div>
    </div>
    <div
      class="warning"
    >
      <div
        class="warning-icon"
      >
        <svg />
      </div>
      <div
        class="warning-message"
      >
        Make sure you have the correct wallet address. There is no way to reverse the transaction.
      </div>
    </div>
    <div
      class="amount"
    >
      <div
        class="input-field form-input"
      >
        <div
          class="label"
        >
          Amount
        </div>
        <div
          class="input-wrapper"
        >
          <input
            class="input"
            name=""
            placeholder="Amount of KOI to send"
            type="text"
            value=""
          />
          <svg
            class="toggle-display"
          />
        </div>
      </div>
      <div
        class="amount-in-usd"
      >
        ~ $ 
        NaN
         USD
      </div>
    </div>
    <button
      class="button-shared button-filled send-button"
    >
      Send KOI
    </button>
  </form>
</div>
`;
